classes/PaymentModule.php

```diff
@@ -986,29 +988,59 @@ protected function createOrderFromCart(
         $order->gift_message = $cart->gift_message;
         $order->mobile_theme = $cart->mobile_theme;
         $order->conversion_rate = $currency->conversion_rate;
-        $amount_paid = !$dont_touch_amount ? Tools::ps_round((float) $amount_paid, Context::getContext()->getComputingPrecision()) : $amount_paid;
+        $amount_paid = !$dont_touch_amount ? Tools::ps_round((float) $amount_paid, $computingPrecision) : $amount_paid;
         $order->total_paid_real = 0;
 
-        $order->total_products = (float) $cart->getOrderTotal(false, Cart::ONLY_PRODUCTS, $order->product_list, $carrierId);
-        $order->total_products_wt = (float) $cart->getOrderTotal(true, Cart::ONLY_PRODUCTS, $order->product_list, $carrierId);
-        $order->total_discounts_tax_excl = (float) abs($cart->getOrderTotal(false, Cart::ONLY_DISCOUNTS, $order->product_list, $carrierId));
-        $order->total_discounts_tax_incl = (float) abs($cart->getOrderTotal(true, Cart::ONLY_DISCOUNTS, $order->product_list, $carrierId));
+        $order->total_products = (float) Tools::ps_round(
+            (float) $cart->getOrderTotal(false, Cart::ONLY_PRODUCTS, $order->product_list, $carrierId),
+            $computingPrecision
+        );
+        $order->total_products_wt = (float) Tools::ps_round(
```

Comment:
why do we need to cast, as ps_round already returns a float ?

Reply:
Initially I did it because I saw in other parts of the code there was a cast.

However indeed `Tools::ps_round()` phpDoc says it returns a float.

When I look at the `ps_round()` code, it seems to rely on ceilf, floorf and `Tools::math_round`
There might be 1 tiny little path in `Tools::math_round` that might return an integer (not sure), all other paths rely on php functions returning floats.

So it seems safe to assert the cast is, indeed, useless 👍

Reply:
Fixed in https://github.com/PrestaShop/PrestaShop/pull/20137/commits/ca6528b2975e891e481b8ee39131171679926d94